http-disk-server

```diff
@@ -0,0 +1,258 @@
+
+
+def close_device(fd):
+    if not fd:
+        return
+
+    while True:
+        try:
+            os.close(fd)
+            return
+        except OSError as e:
+            if e.errno == errno.EBADF:
+                return
+            if e.errno == errno.EINTR:
+                continue
+            print('Cannot close fd {}: `{}`'.format(fd, e))
+            return
+
+
+def get_device_size(fd):
+    disk_capacity = -1
+    while True:
+        try:
+            buf = fcntl.ioctl(fd, BLKGETSIZE64, b' ' * 8)
+            disk_capacity = struct.unpack('L', buf)[0]
+            break
+        except OSError as e:
+            if e.errno != errno.EINTR:
+                print('Cannot get device size: `{}`'.format(e))
+                break
+    return disk_capacity
+
+
+def MakeRequestHandler(disk, disk_capacity):
+    class RequestHandler(BaseHTTPRequestHandler):
+        def __init__(self, *args, **kwargs):
+            # Note: We cannot use this call in python 2:
+            # super(RequestHandler, self).__init__(*args, **kwargs)
+            # > TypeError: must be type, not classobj
+            #
+            # The base class of `BaseHTTPRequestHandler` uses an old def:
+            # "class BaseRequestHandler:"
+            self.disk = disk
+            self.disk_capacity = disk_capacity  # TODO: Detect capacity update?
+            BaseHTTPRequestHandler.__init__(self, *args, **kwargs)
+
+        def parse_range(self):
+            req_range = self.headers.get('range')
+            if req_range is None:
+                self.send_response(400)
+                self.end_headers()
+                return
+
+            if req_range.startswith('bytes'):
+                req_range = req_range[5:].strip()
+                if req_range.startswith('='):
+                    try:
+                        values = req_range[1:].lstrip().split('-')
+                        begin = int(values[0])
+                        end = int(values[1])
+                        if begin <= end and end - begin < self.disk_capacity:
+                            return [begin, end - begin + 1]
+                    except Exception:
+                        pass
+
+            # Return error: "Range Not Satisfiable".
+            self.send_response(416)
+            self.end_headers()
+
+        def parse_content_range(self):
+            req_range = self.headers.get('Content-Range')
+            if req_range is None:
+                self.send_response(400)
+                self.end_headers()
+                return
+
+            if req_range.startswith('bytes'):
+                req_range = req_range[5:].strip()
+                try:
+                    values = req_range.split('-')
+                    begin = int(values[0])
+                    end = int(values[1].split('/')[0].strip())
+                    if begin <= end and end - begin < self.disk_capacity:
+                        return [begin, end - begin + 1]
```

Comment:
No because the parsed values are not exactly the same. The two pieces of code are very similar yes, but I don't want to complexify the code with a factorization.